src/main/java/com/aws/greengrass/componentmanager/plugins/DefaultDockerClient.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.componentmanager.plugins;
+
+import com.aws.greengrass.componentmanager.plugins.exceptions.ImagePullRetryableException;
+import com.aws.greengrass.componentmanager.plugins.exceptions.ImagePullServiceException;
+import com.aws.greengrass.componentmanager.plugins.exceptions.InvalidImageOrAccessDeniedException;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Exec;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Docker CLI wrapper that communicates with Docker Engine to execute user commands.
+ */
+@NoArgsConstructor
+public class DefaultDockerClient {
+    public static final Logger logger = LogManager.getLogger(DefaultDockerClient.class);
+
+    /**
+     * Sanity check for installation.
+     *
+     * @return if docker is installed on the host
+     */
+    boolean dockerInstalled() {
+        CliResponse response = runDockerCmd("docker -v");
+        return response.exit.isPresent() && response.exit.get() == 0;
+    }
+
+    /**
+     * Login to given docker registry.
+     *
+     * @param registry Registry to log into, with credentials encapsulated
+     * @return if docker login was successful
+     * @throws ImagePullServiceException Error in authenticating with the registry
+     */
+    boolean login(Registry registry) throws ImagePullServiceException {
+        // TODO : [Blocker] Ensure credentials in command to be run do not get logged
+        CliResponse response = runDockerCmd(String.format("docker login %s -u %s -p %s", registry.getRegistryId(),
+                registry.getCredentials().getUsername(), registry.getCredentials().getPassword()));
+        if (response.exit.isPresent()) {
+            if (response.exit.get() != 0) {
+                throw new ImagePullServiceException(
+                        String.format("Error logging into the registry using credentials - %s", response.out));
+            }
+        }
+        return true;
```

Comment:
Do we need both the boolean return value and the exceptions?
If `response.exit` is not present, is login still successful?

Reply:
We don't need it, it's just that RetryUtils is written in way it requires a return value. I will spend some time to fix it after other issues are resolved